app/models/service.rb

```diff
@@ -97,11 +97,11 @@ def direct_vms
   end
 
   def all_vms
-    MiqPreloader.preload_and_map(subtree, :direct_vms)
+    MiqPreloader.preload_and_map(subtree, :vms)
```

Comment:
I think sub tree direct_vms is right.

For every mode in the tree, load only it's own vms.

This works because every mode in the tree is a single call. Then the in clause is one more.

Did the old code perform an N+1?